examples/spec/integration/register_namespace_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'workflows/timeout_workflow'
+
+describe 'Temporal.register_namespace' do
+  it 'can register a new namespace' do
+    # have to generate a new namespace on each run because currently can't delete namespaces
+    name = "test_namespace_#{SecureRandom.uuid}"
+    description = 'this is the description'
+    retention_period_days = 30
+    data = { test: 'value' }
+
+    Temporal.register_namespace(name, description, retention_period_days: retention_period_days, data: data)
+
+    # fetch the namespace from Temporal and check it exists and has the correct settings 
+    # (need to wait a few seconds for temporal to catch up so try a few times)
+    attempts = 0
+    while attempts < 30 do
```

Comment:
Are we sure 15s is enough? Also, how long does it normally take?

Reply:
normally takes 3-5seconds and i ran this like 100 times and it didn't fail!